### DISCUSSION TITLE
Waiting for SetHttpClientTimeout bug to be solved. This is costing me money!

### DISCUSSION THREAD

**Discussion by bmaluijb** [deepgram/community#554] [General help] [answered]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

Hi, 

My users are using audio files with very long durations (some longer then 6 hours). I run [into this bug,](https://github.com/deepgram/deepgram-dotnet-sdk/issues/127) which makes my request time out, and makes my app try again. That costs money! I don't want to change my architecture to listen for the results. I just want to use the SetHttpClientTimeout in the .NET SDK. There is already a PR for the fix, but t looks like it is stuck: https://github.com/deepgram/deepgram-dotnet-sdk/pull/179 Please merge this and release a new version of the .NET SDK asap.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment]
I am just coming on board to the project and will be taking a look at this today along with other .NET related stuff. From a quick glance, it looks like it isn't much of a fix but more trying to undo a mistake. We need to verify and perform due diligence before committing any changes to prevent any unintended consequences of the change. If you rollback to .NET [v3.1.0](https://github.com/deepgram/deepgram-dotnet-sdk/releases/tag/3.1.0) that should remove the timeout functionality and get you going again.

**Reply by bmaluijb** [reply]
Thanks! As a work-around, I downloaded the repo and run it locally, with the fixed SetHttpClientTimeout functionality, as I really need it,

**Reply by davidvonthenen** [reply]
It's being worked on.

**Comment by davidvonthenen** [comment, marked as answer]
This was released and includes a fix for the issue. Details here:
https://github.com/deepgram/deepgram-dotnet-sdk/pull/207#issuecomment-1927872852

Release:
https://github.com/deepgram/deepgram-dotnet-sdk/releases/tag/3.4.1
